lib/rubocop/cop/performance/deterministic_regexp.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop identifies places where `split` argument can be replaced from
+      # a deterministic regexp to a string.
+      #
+      # @example
+      #   # bad
+      #   'a,b,c'.split(/,/)
+      #
+      #   # good
+      #   'a,b,c'.split(',')
+      class DeterministicRegexp < Base
+        extend AutoCorrector
+
+        MSG = 'Use string as argument instead of regexp.'
+        RESTRICT_ON_SEND = %i[split].freeze
+        DETERMINISTIC_REGEX = /\A(?:#{LITERAL_REGEX})+\Z/.freeze
+
+        def_node_matcher :split_call_with_regexp?, <<~PATTERN
+          {(send !nil? :split {regexp})}
+        PATTERN
+
+        def on_send(node)
+          return unless split_call_with_regexp?(node)
+          return unless determinist_regexp?(node.first_argument)
+
+          add_offense(node) do |corrector|
+            autocorrect(corrector, node)
+          end
+        end
+
+        private
+
+        def determinist_regexp?(first_argument)
+          first_argument.source =~ DETERMINISTIC_REGEX
```

Comment:
Can you use `match?` method instead of?

```suggestion
          DETERMINISTIC_REGEX.match?(first_argument.source)
```